internal/plan/plan.go

```diff
@@ -178,6 +184,27 @@ func (s *Service) Merge(other *Service) {
 	if other.BackoffLimit.IsSet {
 		s.BackoffLimit = other.BackoffLimit
 	}
+	s.LogTargets = dedupStrings(s.LogTargets, other.LogTargets)
+}
+
+// dedupStrings merges arr1 and arr2 with deduplication (removing any duplicate
+// entries in the merged slice).
+func dedupStrings(arr1, arr2 []string) []string {
+	seen := map[string]struct{}{}
+	var merged []string
+
+	for _, s := range arr1 {
+		if _, ok := seen[s]; !ok {
+			merged = append(merged, s)
+		}
+	}
+	for _, s := range arr2 {
+		if _, ok := seen[s]; !ok {
+			merged = append(merged, s)
+		}
+	}
```

Comment:
Can we please change this function so it looks like this:
```
// appendUnique appends into a the elements from b which are not yet present
// and returns the modified slice.
func appendUnique(a []string, b ...string) []string {
Outer:
	for _, bn := range b {
		for _, an := range a {
			if an == bn {
				continue Outer
			}
		}
		a = append(a, bn)
	}
	return a
}
```

Besides being shorter, this is significantly faster for cases we care about, uses only the minimum required memory, and does not allocate at all when there's nothing to do. The signature also matches that of `append`, which makes adding single strings slightly nicer and the behavior more expectable.

After this PR series is done, it'd be nice to move this into x-go/strutil.